src/librustc_codegen_ironox/lib.rs

```diff
@@ -0,0 +1,132 @@
+use rustc::ty::{self, TyCtxt};
+use rustc_codegen_utils::codegen_backend::CodegenBackend;
+use rustc_codegen_utils::target_features::{all_known_features, X86_WHITELIST};
+use rustc_data_structures::sync::Lrc;
+use rustc_mir::monomorphize::collector;
+use rustc_mir::monomorphize::item::MonoItem;
+
+mod back {
+    pub use rustc_codegen_utils::symbol_export;
+    pub use rustc_codegen_utils::symbol_names;
+}
+
+pub struct IronOxCodegenBackend(());
+
+impl !Send for IronOxCodegenBackend {}
+impl !Sync for IronOxCodegenBackend {}
+
+pub fn target_feature_whitelist(sess: &Session)
+    -> &'static [(&'static str, Option<&'static str>)] {
+    match &*sess.target.target.arch {
+        "x86" | "x86_64" => X86_WHITELIST,
+        _ => &[],
+    }
+}
+
+impl IronOxCodegenBackend {
+    pub fn new() -> Box<dyn CodegenBackend> {
+        box IronOxCodegenBackend(())
+    }
+}
+
+impl CodegenBackend for IronOxCodegenBackend {
+    fn init(&self, _sess: &Session) {}
+
+    fn print(&self, _req: PrintRequest, _sess: &Session) {}
+
+    fn print_passes(&self) {}
+
+    fn print_version(&self) {}
+
+    fn metadata_loader(&self) -> Box<dyn MetadataLoader + Sync> {
+        box metadata::IronOxMetadataLoader
+    }
+
+    fn provide(&self, providers: &mut ty::query::Providers) {
+        providers.target_features_whitelist = |tcx, cnum| {
+            assert_eq!(cnum, LOCAL_CRATE);
+            if tcx.sess.opts.actually_rustdoc {
+                Lrc::new(
+                    all_known_features()
+                        .map(|(a, b)| (a.to_string(), b.map(|s| s.to_string())))
+                        .collect(),
+                )
+            } else {
+                Lrc::new(
+                    target_feature_whitelist(tcx.sess)
+                        .iter()
+                        .map(|&(a, b)| (a.to_string(), b.map(|s| s.to_string())))
+                        .collect(),
+                )
+            }
+        };
+        back::symbol_names::provide(providers);
+        back::symbol_export::provide(providers);
+    }
+
+    fn provide_extern(&self, providers: &mut ty::query::Providers) {
+        back::symbol_export::provide_extern(providers);
+    }
+
+    fn codegen_crate<'a, 'tcx>(
+        &self,
+        tcx: TyCtxt<'a, 'tcx, 'tcx>,
+        _rx: mpsc::Receiver<Box<dyn Any + Send>>,
+    ) -> Box<dyn Any> {
+        // Print the MIR
+        for mono_item in collector::collect_crate_mono_items(
+            tcx, collector::MonoItemCollectionMode::Eager).0 {
+            match mono_item {
+                MonoItem::Fn(inst) => {
+                    let def_id = inst.def_id();
+                    eprintln!("Def ID: {:?}", def_id);
+                    let mir = tcx.instance_mir(inst.def);
+                    for bb in mir.basic_blocks() {
+                        eprintln!("Statements: {:?}", bb.statements);
+                    }
+                    if def_id.is_local() {
+                        let _ = inst.def.is_inline(tcx);
```

Comment:
I know this is a stub, but I wondered if this thrown-away-assignment is deliberate?

Reply:
No, the entire block doesn't need to be there. I'll remove it.